packages/smithy-client/src/parse-utils.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Asserts a value is a boolean and returns it.
+ *
+ * @param value A value that is expected to be a boolean.
+ * @returns The value if it is a boolean, otherwise an error is thrown.
+ */
+export function expectBoolean(value: any): boolean {
```

Comment:
it seems like a lot of the deserialization code would be cleaner if these expectation functions returned `undefined` if `value` was `undefined` or `null` -- this could reduce the amount of code overall instead of adding to it.

Reply:
You are correct. Went from adding about 20k lines to removing about 20k lines